patterns/singleton.py

```diff
@@ -0,0 +1,17 @@
+"""
+Singleton Pattern Example
+Ensures a class has only one instance and provides a global point of access to it.
```

Comment:
In line with doctoring standards - could the second line of this doctoring be empty please. (add empty line)

See https://github.com/faif/python-patterns/blob/master/patterns/structural/facade.py for an example of a good docString. Unfortunately there are many poor examples around.